chore: document new change detection behavior. (#103)
terramate-io/terramate-docs: cli/change-detection/index.md

terramate list --changed
```

Since `v0.11.0` untracked and uncommitted files are also considered changes.

::: info

> _Untracked file_ means Git doesn't know about its existence.

Reviewer: ```suggestion
_Untracked file_ is a file which isn't tracked by the Git index
```

> _Uncommitted file_ means Git knows about them but the latest changes in them are not committed to Git yet.

Reviewer: ```suggestion
_Uncommitted file_ is a file tracked by the Git index, but its latest changes are not committed to Git yet
```

terramate list --changed
```

Since `v0.11.0` untracked and uncommitted files are also considered changes.

::: info

> _Untracked file_ means Git doesn't know about its existence.

Reviewer: ```suggestion
An _Untracked file_ is a file that Git doesn't recongnize.
```

Author: I think "Git doesn't recognize" is not correct.
Other ways of saying:

- Untracked are files not added nor committed
- Untracked are files not tracked by Git index
- Untracked are files that have been created within your repo's working directory but have not yet been added to the repository's tracking index using the `git add` command.  (got this in the [Atlassian docs](https://www.atlassian.com/git/tutorials/undoing-changes/git-clean#:~:text=Untracked%20files%20are%20files%20that,using%20the%20git%20add%20command.)

what do you prefer?

Reviewer: I think the second option is good: clear & concise.

> _Uncommitted file_ means Git knows about them but the latest changes in them are not committed to Git yet.

Reviewer: ```suggestion
_Uncommitted file_ is a file where changes are recognized by Git but not yet committed.
```

Author: this is not much accurate either. Check my other answer.

:::

The change detection behavior can be customized in the configuration or by command-line flags.

Example:

```hcl
terramate {
  config {
    change_detection {
      git {
        # valid options: true, "on", false, "off"
        untracked   = "off"
        uncommitted = "off"
      }
    }
  }
}
```

> The `--enable-change-detection=<options>` and `--disable-change-detection=<options>` can be used to override
> the default and the configuration. These flags are supported in the commands below (check their documentation for usage):

Reviewer: ```suggestion
Use `--enable-change-detection=<options>` and `--disable-change-detection=<options>` to override the default settings and configuration. These flags are supported in the commands listed below (refer to their documentation for usage):
```
